lib/helpers/doChannelReport.js

```diff
@@ -1,192 +1,39 @@
 'use strict';
 
-var log = require('../../getLogger')('channel report');
-var async = require('async');
-var _ = require('underscore');
-// var moment = require('moment');
-var models = require('../../models');
-var timeHelper = require('./time');
-var standupHelper = require('./getStandupReport');
-
-module.exports = function doChannelReport(bot, reportChannel, update, userRealName) {
-  var date = timeHelper.getCurrentDate();
-  var teamDomain;
-  if (update) {
-    bot.api.team.info({}, function(err, response) {
-      if (err) {
-        console.log(err);
-      } else {
-        teamDomain = response.team.domain;
-      }
-    });
-  }
-
-  models.Channel.findOne({
-    where: {
-      name: reportChannel
-    }
-  }).then(function(channel) {
-
-    // Set report audience
-    // var audience = channel.audience === null ? '<!here>' : channel.audience;
-    // if (audience.search(/<!/) !== 0) {
-    //   audience = '@'+audience;
-    // }
-
-    // Find standup messages for that channel & for today
-    models.Standup.findAll({
-      where: {
-        channel: channel.name,
-        date: date
-      }
-    }).then(function (standups) {
-      // Begin a Slack message for this channel
-      // https://api.slack.com/docs/attachments
-      var report = {
-        attachments: [],
-        channel: channel.name
-      };
-      var attachments = [];
+const async               = require('async');
+const models              = require('../../models');
+const timeHelper          = require('./time');
+const createNewChannelReport  = require('./createNewChannelReport');
+const updateChannelReport     = require('./updateChannelReport');
+
+module.exports = function doChannelReport(bot, channelName, update, userName, actionCallback) {
```

Comment:
There are a few async deadends in the code that make that hard right now. I would definitely switch to promises, and doing it for this batch of commits would have been my next step.

I think first though all the async has to be linked together, instead of occasional functions that are orphaned in async space.